src/libcore/str/mod.rs

```diff
@@ -822,6 +823,13 @@ impl Iterator for Bytes<'_> {
     {
         self.0.rposition(predicate)
     }
+
+    #[inline]
+    unsafe fn get_unchecked(&mut self, idx: usize) -> u8 {
+        // SAFETY: we directly wrap `self.0`, so all safety requirements are
```

Comment:
(note: not something that needs to be fixed, just me rambling) I find this comment interesting. I understand the need to add such a comment around an `unsafe { }` block as that's where callers spend effort to satisfy the invariants of the unsafe callee, however in this case, as this comment alludes to, the responsibility for all the invariants is propagated to the caller of the wrapping function… so this comment doesn’t appear to be adding any value. If anything I’d expect a doc-comment on top of this wrapper function specifying that the caller should look at docs of `$ANOTHER_FUNCTION` for invariants they need to uphold.